python/triton/runtime/redis.py

```diff
@@ -0,0 +1,31 @@
+import os
+from typing import Dict, List
+
+from .cache import RemoteCacheBackend
+
+import redis
```

Comment:
can we import this lazily so we Triton doesn't have to depend on redis?

Reply:
Ah yeah, I was wondering this at the time.  I think the only way this file would get loaded is if the user sets e.g. `TRITON_CACHE_MANAGER=triton.runtime.cache:RemoteCacheManager TRITON_REMOTE_CACHE_BACKEND=triton.runtime.redis:RedisRemoteCacheBackend`, but I didn't know if there was potentially some other setup that was sensitive to top-level imports even the module itself isn't explicitly loaded.

Reply:
Updated to do this.